napari/components/add_layers_mixin.py

```diff
@@ -150,6 +152,17 @@ def add_image(
             Translation values for the layer. If a list then must be a list of
             tuples of float with the same length as the axis that is being
             expanded as channels.
+        rotate : float, 3-tuple of float, n-D array or list.
+            If a float convert into a 2D rotate matrix using that value as an
+            angle. If 3-tuple convert into a 3D rotate matrix, rolling a yaw,
+            pitch, roll convention. Otherwise assume an nD rotate. Angle
+            conversion are done either using degrees or radians depending on the
+            degrees boolean parameter. If a list then must have same length as
+            the axis that is being expanded as channels.
+        shear : 1-D array or list.
```

Comment:
It was not clear to me how to use the shear parameter, or even how many values should be in the shear list I pass in. When I read this docstring I think there should be as many elements in the shear list as there are channels in the image, but this isn't true for RGB images like the example cat image `skimage.data.chelsea()`.

The error message I saw while playing around with this "ValueError: 2 is a strange number of shear elements" told me what I'd done wrong, but not necessarily how to fix it (I tried a bunch of random values to work this out instead).

It's also weird to me that in cases where shear is expecting a list containing a single value, I can't just pass in that value as a scalar. That seems like maybe something that could be improved for convenience, although further tinkering reveals that I can't say `scale=2` (or even `scale=[2]`) and have all axes scaled by the same value as well. So perhaps I shouldn't expect to be able to do this at all.

Reply:
> It was not clear to me how to use the shear parameter, or even how many values should be in the shear list I pass in.

Hmm yeah, this is maybe not ideal. So right now shear is the values of an upper triangular matrix in a vector, so for a 2D case that is one value, a 3D case that is 3 values, a 4D case that is 6 values. The wikipedia page on the [shear matrix](https://en.wikipedia.org/wiki/Shear_matrix) could maybe do with some work but it covers some.

An alternative would be to accept the full nD matrix and check it is upper triangular. I'm curious how people who work with skewed data normally represent the shear (say from a lattice-lightsheet).

> It's also weird to me that in cases where shear is expecting a list containing a single value, I can't just pass in that value as a scalar. 

Right now you can't do `shear=3` in the 2D case, you have to do `shear=[3]`, but we could change that.

> although further tinkering reveals that I can't say scale=2 (or even scale=[2]) and have all axes scaled by the same value as well.

Yeah, that we should maybe change - i.e. if you pass a float we broadcast to all dimensions that value. People would might like that

Reply:
The shear is a bit awkward. I expect there's a construction along the lines of "for each 1 unit moved along dir_1, move a units in dir_2 and b units in dir_3". This is semi-intuitive for shears where dir_1 is one of your axes -- and I'm guessing that dir_1 = z is the most common and practical case - and probably a little funky otherwise.

More generally, my feeling when I played with this before is that (at least in 3d) this is a feature that would really benefit from some kind of preview dialog popping up. Thinking in Euler angles and upper triangular matrix elements is not super natural to me and probably less so to much of the intended audience. If an image pyramid exists for the current image, I could imagine actually using that for preview. Otherwise, showing something like how the coordinate axes (or a unit cube) transform under the proposed parameters could be really helpful. (A related question: is there any kind of overlay in 3d rendering that shows what the current orientation of the coordinate axes are?)

Reply:
Those are good points @bkcooper

> (A related question: is there any kind of overlay in 3d rendering that shows what the current orientation of the coordinate axes are?)

vispy has one of these, which we're not using now, but I think it will make sense to either provide our own or use theirs. we could actually do that in a separate PR. I agree it might help with this!!

Reply:
We may want a new example file with different setups to illustrate how to use the newly added rotate/sheer/affine transformation, the sudoku example from earlier is a great demo in my opinion. This example can help document what/how to set parameters and help define what we intend to expose and support